--- distrib/amd64/common/install.md.orig	Mon May  5 13:00:43 2014
+++ distrib/amd64/common/install.md	Mon May  5 20:27:14 2014
@@ -98,6 +98,10 @@
 
 	_f=/tmp/fstab.$_disk
 	if [[ $_disk == $ROOTDISK ]]; then
+		if [[ -s /tmp/disklabel.$_disk ]]; then
+			disklabel $FSTABFLAG $_f -E $_disk < /tmp/disklabel.$_disk >/dev/null
+			return
+		fi
 		while :; do
 			echo "The auto-allocated layout for $_disk is:"
 			disklabel -h -A $_disk | egrep "^#  |^  [a-p]:"
--- distrib/amd64/common/list.orig	Mon May  5 16:53:18 2014
+++ distrib/amd64/common/list	Mon May  5 16:53:19 2014
@@ -81,4 +81,6 @@
 SCRIPT	${CURDIR}/../../miniroot/upgrade.sh	upgrade
 SCRIPT	${CURDIR}/../../miniroot/install.sh	install
 SCRIPT	${CURDIR}/../../miniroot/install.sub	install.sub
+SCRIPT	${CURDIR}/../../miniroot/install.pre	install.pre
+SCRIPT	${CURDIR}/../../miniroot/install.post	install.post
 SPECIAL	chmod 755 install upgrade
--- distrib/amd64/ramdisk_cd/list.local.orig	Mon May  5 14:59:14 2014
+++ distrib/amd64/ramdisk_cd/list.local	Mon May  5 14:59:14 2014
@@ -10,6 +10,7 @@
 LINK	instbin					sbin/mount_ext2fs
 LINK	instbin					sbin/mount_msdos
 LINK	instbin					sbin/mount_udf
+LINK	instbin					sbin/mount_tmpfs
 
 COPY	${DESTDIR}/etc/firmware/kue		etc/firmware/kue
 
--- distrib/miniroot/install.sh.orig	Fri Feb 21 19:11:02 2014
+++ distrib/miniroot/install.sh	Mon May  5 20:42:49 2014
@@ -99,6 +99,7 @@
 	# /tmp/fstab.$DISK is created here with 'disklabel -f' or
 	# 'disklabel -F' depending on the value of $FSTABFLAG.
 	rm -f /tmp/*.$DISK
+	pre_disklabel
 	md_prep_disklabel $DISK || { DISK= ; continue ; }
 
 	# Make sure there is a '/' mount point.
@@ -317,6 +318,9 @@
 		/mnt/etc/sysctl.conf >/tmp/sysctl.conf
 	cp /tmp/sysctl.conf /mnt/etc/sysctl.conf
 fi
+
+echo "done."
+. install.post
 
 # Perform final steps common to both an install and an upgrade.
 finish_up
--- distrib/miniroot/install.sub.orig	Sun Mar  2 16:12:16 2014
+++ distrib/miniroot/install.sub	Mon May  5 19:35:00 2014
@@ -88,6 +88,7 @@
 #	MDXAPERTURE - set machdep.allowaperture=value in sysctl.conf
 #	NCPU	    - the number of cpus for mp capable arches
 . install.md
+. install.pre
 
 set_term() {
 	local _layouts
@@ -2261,6 +2262,10 @@
 	local _f _ifdev _mac _mode _server _lf
 	[[ -x /sbin/dhclient ]] || return
 
+	action=install
+	pre_prepconf
+	return 0
+
 	_ifdev=$(get_ifdevs netboot)
 	if [[ -z $_ifdev ]]; then
 		set -- $(get_ifdevs)
@@ -2353,6 +2358,9 @@
 	donetconfig
 
 	((nifs != 0)) && startftplist
+
+	echo
+	pre_preserve
 
 	echo
 	while :; do
--- distrib/miniroot/dot.profile.orig	Fri Feb 21 18:30:14 2014
+++ distrib/miniroot/dot.profile	Mon May  5 15:04:02 2014
@@ -69,7 +69,7 @@
 	# after a timeout, but only the very first time around.
 	timeout=false
 	timer_pid=
-	if [ ! -f /tmp/noai ] && ifconfig netboot >/dev/null 2>&1; then
+	if [ ! -f /tmp/noai ]; then # && ifconfig netboot >/dev/null 2>&1; then
 		echo "Starting non-interactive mode in 5 seconds..."
 		>/tmp/noai
 
--- sys/arch/amd64/conf/RAMDISK_CD.orig	Wed Feb 19 07:32:57 2014
+++ sys/arch/amd64/conf/RAMDISK_CD	Mon May  5 14:47:19 2014
@@ -22,6 +22,9 @@
 option		UDF
 option		MSDOSFS		# MS-DOS file system
 
+option		DIAGNOSTIC
+option		TMPFS
+
 option		INET		# IP + ICMP + TCP + UDP
 option		INET6		# IPv6 (needs INET)
 
